web/client-api/src/main/java/io/deephaven/web/client/api/input/JsInputTable.java

```diff
@@ -71,14 +73,15 @@ public String[] getKeys() {
 
     /**
      * A list of the key Column objects
-     * 
+     *
      * @return {@link Column} array.
      */
     @JsProperty
-    public Column[] getKeyColumns() {
-        return table.findColumns(keys);
+    public JsArray<Column> getKeyColumns() {
+        return Js.uncheckedCast(table.lastVisibleState().getKeyColumns());
```

Comment:
The goal of this patch is not to correct behavior here, but to make dh.InputTable.keyColumns always return the same instance of the array. Instead of making all CTS instances store the array as well, why not just memoize the result here - or assign it in the input table constructor.

For example, one more line in the ctor could be:
```java
this.keyColumns = JsObject.freeze(table.findColumns(keys));
```
or 
```java
this.keyColumns = JsObject.freeze(table.getColumns().filter((c, i) -> c.isInputTableKeyColumn()));
```
Using `Column[]` or `JsArray<Column>` are the same as far as Js is concerned, only matters a little in JS.